### PR TITLE
Adding GetObjectAttributes S3 action to our policies list

### PR DESCRIPTION
**Pull request by zveinn**

### CODE REVIEW

**Review by klauspost — APPROVED** [review]
LGTM

**Comment by klauspost** [comment]
Seems like [CI Go version](https://github.com/minio/pkg/blob/main/.github/workflows/vulncheck.yml#L17) needs a bump along with github.com/lestrrat-go/jwx

**Comment by zveinn** [comment]
> Seems like [CI Go version](https://github.com/minio/pkg/blob/main/.github/workflows/vulncheck.yml#L17) needs a bump along with github.com/lestrrat-go/jwx

yea, there is another PR to fix the tests. If they are still failing when I'm about to be done with this I'll grab that one too.

**Review by harshavardhana — APPROVED** [review]

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Actually this is not sufficient @zveinn you need to add this into supported actions list and also add relevant headers that are valid. 

Look for other actions where all they are referenced check `GetObjectAction`

**Comment by zveinn** [comment]
> Actually this is not sufficient @zveinn you need to add this into supported actions list and also add relevant headers that are valid.
> 
> Look for other actions where all they are referenced check `GetObjectAction`

Will do

File: policy/action.go [hunk 1]
```diff
@@ -372,7 +374,9 @@ func createActionConditionKeyMap() ActionConditionKeyMap {
 				condition.ExistingObjectTag.ToKey(),
 			}, commonKeys...)...),
 
-		HeadBucketAction: condition.NewKeySet(commonKeys...),
+		HeadBucketAction:                 condition.NewKeySet(commonKeys...),
+		GetObjectAttributesAction:        condition.NewKeySet(commonKeys...),
```
Comment (harshavardhana) [inline policy/action.go:378]: https://docs.aws.amazon.com/AmazonS3/latest/userguide/list_amazons3.html Actually this supports tags as well so you need to add 

```
		GetObjectAttributesAction: condition.NewKeySet(
			append([]condition.Key{
				condition.S3VersionID.ToKey(),
				condition.ExistingObjectTag.ToKey(),
			}, commonKeys...)...),
```

Not just commonKeys.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: policy/action.go (hunk 1 above)
Comment (zveinn) [reply policy/action.go]: GetObjectAttributes did not require the versionID, the GetObjectVersionAttributes was the one. 

But they've been added now.

**Review by klauspost — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**